keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,616 @@
+
+This KEP is intentionally scoped small, and intends to keep the scope just to `ownerReferences` and not to other fields of the owner object to prevent security risks.
+
+## Motivation
+
+DaemonSets and ReplicaSets (via Deployments) can dynamically orphan and adopt pods. Consider a scenario where, on day 1, a pod named foobar-deadbeef93-xyz76 is owned by the ReplicaSet foobar-deadbeef93, which in turn is managed by the Deployment foobar. After a node restart, the pod might be replaced by foobar-deadbeef93-abc12, orphaning the original pod foobar-deadbeef93-xyz76 and potentially causing unwanted behavior. This issue arises when pods create resources like ConfigMaps or CustomResources and these resources are not correctly reassigned to the new pod, leading to orphaned objects. Ensuring that pods can pass their ownerReferences to new objects they manage would prevent this issue, as the ownership hierarchy would be preserved even when pods are recreated.
+
+For example, a CustomResource created by a pod managed by a DaemonSet may be unexpectedly garbage collected if the pod is deleted. This can disrupt system behavior, as the necessary resource is lost. By allowing the pod to inherit and pass down ownerReferences, the CustomResource would remain correctly managed by the DaemonSet, avoiding such disruptions.
+
+### Goals
+
+* Gain access to pod `ownerReferences` on pods through volume mounts.
+* Gain access to pod `ownerReferences` on pods through environmental variables.
+
+### Non-Goals
+
+- Not to expose additional pod info outside of `metadata.ownerReferences`.
+
+## Proposal
+
+The initial design includes:
+
+* Being able to pass pod `ownerReferences` to volumes with fieldPath of `metadata.ownerReferences`.
+* Being able to pass pod `ownerReferences` to environmental variables with fieldPath of `metadata.ownerReferences`.
+
+### User Stories (Optional)
+
+#### Story 1
+xref https://github.com/kubernetes-sigs/node-feature-discovery/issues/1752
+
+CustomResource object created by a pod owned by a DaemonSet is garbage collected when the pod is deleted by unexpected reasons. This triggers unwanted behavior in the system.
+If the CustomResource object inherits the ownerReferences from the pod, it will be garbage collected only when the DaemonSet is deleted. This will add extra reliability to the system by projecting the ownerReferences from the Deployment/DaemonSet/StatefulSet to the CustomResource object.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- Security Risk: Some environments may not want to expose the ownerReferences of a pod to the pod itself.
+  - Mitigation: This feature could be added as a feature gate, disabled by default.
```

Comment:
This is not a mitigation to this KEP.  If there is some cluster wherein the admin DOES NOT want pods to be able to know their owners, the mitigation would be to install an admission check (webhook or Validating Admission Policy) to reject pods that try to use this.

Reply:
I have edited the Mitigation entry, thanks @thockin